cli/src/commands/developer/deploy.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::{CurrentNetwork, Developer};
+
+use snarkvm::prelude::{
+    ConsensusMemory,
+    ConsensusStore,
+    Plaintext,
+    PrivateKey,
+    ProgramID,
+    Query,
+    Record,
+    ToBytes,
+    Transaction,
+    VM,
+};
+
+use anyhow::{ensure, Result};
+use clap::Parser;
+use colored::Colorize;
+use std::{path::PathBuf, str::FromStr};
+
+/// Deploys an Aleo program.
+#[derive(Debug, Parser)]
+pub struct Deploy {
+    /// The name of the program to deploy.
+    #[clap(parse(try_from_str), help = "The ID of the program to deploy")]
+    program_id: ProgramID<CurrentNetwork>,
+    /// A path to a directory containing a manifest file. Defaults to the current working directory.
+    #[clap(long, help = "A path to a directory containing a manifest file")]
+    path: Option<String>,
+    /// The private key used to generate the deployment.
+    #[clap(short = 'p', long, help = "The private key used to generate the deployment")]
+    private_key: String,
+    /// The endpoint to query node state from.
+    #[clap(short = 'q', long, help = "The endpoint to query node state from")]
+    query: String,
+    // TODO (raychu86): Update the default.
+    /// The deployment fee in gates.
+    #[clap(short, long, help = "The deployment fee in gates, defaults to 0")]
+    fee: Option<u64>,
+    /// The record to spend the fee from.
+    #[clap(short, long, help = "The record to spend the fee from")]
+    record: String,
+    /// Display the generated transaction.
+    #[clap(
+        short,
+        long,
+        default_value = "false",
+        help = "Display the generated transaction",
+        conflicts_with = "broadcast"
+    )]
+    display: bool,
```

Comment:
Bool params in clap automatically default to false.

```suggestion
    /// Display the generated transaction.
    #[clap(
        short,
        long,
        conflicts_with = "broadcast"
    )]
    display: bool,
```